airflow/task/task_runner/base_task_runner.py

```diff
@@ -54,32 +52,24 @@ def __init__(self, local_task_job):
             except conf.AirflowConfigException:
                 self.run_as_user = None
 
+        # Always provide a copy of the configuration file settings
+        cfg_path = tmp_configuration_copy()
```

Comment:
What cleans this up? If it's not cleaned up, seems like a good place for a context manager.

Reply:
I tried using a context manager, but this file is actually removed by the receiving new process so triggered by the task runner: https://github.com/apache/incubator-airflow/blob/master/airflow/bin/cli.py#L395 so, the context does not work for this case.
This is usually a bad pattern because the `run` command in the `cli` (a new process) is removing a file it does not maintain, making it harder for the caller (the parent process) to clean up the files the it actually creates / maintains. Fixing that design is out of scope of this PR.

Reply:
@edgarRd can you at least create a JIRA for that issue?

Reply:
@bolkedebruin https://issues.apache.org/jira/browse/AIRFLOW-2084